### PR TITLE
CVSL-1899 Add test and fix issue with job

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: jobs/promptLicenceCreationService.test.ts [hunk 1]
```diff
@@ -124,15 +125,154 @@ describe('prompt licence creation service ', () => {
           urgentPromptCases: [{ crn: undefined, name: 'aaa', releaseDate: '2023-01-02' }],
         },
       ] as EmailContact[]
-      await promptLicenceCreationService.notifyComsToPromptLicenceCreation(expectedRequest)
+      await promptLicenceCreationService.notifyComOfUpcomingReleases(expectedRequest)
       expect(licenceApiClient.notifyComsToPromptEmailCreation).toHaveBeenCalledWith(expectedRequest)
     })
+
+    it('should filter out email groups that have no cases', async () => {
```
Comment (andrewrlee) [inline jobs/promptLicenceCreationService.test.ts:132]: missing test

File: jobs/promptLicenceCreationService.test.ts [hunk 2]
```diff
@@ -124,15 +125,154 @@ describe('prompt licence creation service ', () => {
...
+      ] as EmailContact[]
+      await promptLicenceCreationService.notifyComOfUpcomingReleases(cases)
+      const [a, , b] = cases
+      expect(licenceApiClient.notifyComsToPromptEmailCreation).toHaveBeenCalledWith([a, b])
+    })
   })
 })
 
-function createManagedCase(offenderManagers: OffenderManager[], licenceStatus: LicenceStatus): ManagedCase {
+describe('buildEmailGroups', () => {
```
Comment (andrewrlee) [inline jobs/promptLicenceCreationService.test.ts:160]: test to capture bug

File: jobs/promptLicenceCreationService.ts [hunk 3]
```diff
@@ -47,10 +47,10 @@ export default class PromptLicenceCreationService {
   ): Promise<EmailContact[]> => {
     const managedCases = [...urgentPromptCases, ...initialPromptCases]
 
-    const mapPrisonerToReleaseCase = (prisoner: CvlPrisoner) => {
+    const mapPrisonerToReleaseCase = (prisoner: CvlPrisoner & { crn: string }) => {
       return {
         name: convertToTitleCase(`${prisoner.firstName} ${prisoner.lastName}`),
-        crn: undefined as string, // FIXME - this is never worked!
+        crn: prisoner.crn,
```
Comment (andrewrlee) [inline jobs/promptLicenceCreationService.ts:53]: We enrich the prisoner data before this so crn is available - its just the types were incorrect and a bit bodged previously

**Review by nishanthmahasamudram — APPROVED** [review]
LGTM

**Pull request merged**